Build failure: freeipmi and nut after staging-next 2024-12-04 #361878

## Steps To Reproduce

Steps to reproduce the behavior:

1. build freeipmi and nut at nixpkgs/master

## Build log

```
error: builder for '/nix/store/8f4qj079jqpsp420hmf0slqlhdb58xnb-freeipmi-1.6.14.drv' failed with exit code 2;
       last 25 log lines:
       > libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../config -I../common/debugutil -I../common/miscutil -I../common/portability -I../libfreeipmi/include -I../libfreeipmi/include -I../libfreeipmi -DIPMI_IPCKEY=\"/nix/store/0lnd5a6imppj11arqipq6vpkkk90ish4-freeipmi-1.6.14/var/lib/freeipmi/ipckey\" -DIPMI_DEBUG_IPCKEY=\"../libfreeipmi/driver/ipmi-semaphores.h\" -D_REENTRANT -g -O2 -c sel/ipmi-sel-string-linux-kernel.c  -fPIC -DPIC -o sel/.libs/libfreeipmi_la-ipmi-sel-string-linux-kernel.o
       > /nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../config  -I../common/debugutil -I../common/miscutil -I../common/portability -I../libfreeipmi/include -I../libfreeipmi/include -I../libfreeipmi -DIPMI_IPCKEY=\"/nix/store/0lnd5a6imppj11arqipq6vpkkk90ish4-freeipmi-1.6.14/var/lib/freeipmi/ipckey\" -DIPMI_DEBUG_IPCKEY=\"../libfreeipmi/driver/ipmi-semaphores.h\" -D_REENTRANT   -g -O2 -c -o sel/libfreeipmi_la-ipmi-sel-string-quanta-s99q.lo `test -f 'sel/ipmi-sel-string-quanta-s99q.c' || echo './'`sel/ipmi-sel-string-quanta-s99q.c
       > libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../config -I../common/debugutil -I../common/miscutil -I../common/portability -I../libfreeipmi/include -I../libfreeipmi/include -I../libfreeipmi -DIPMI_IPCKEY=\"/nix/store/0lnd5a6imppj11arqipq6vpkkk90ish4-freeipmi-1.6.14/var/lib/freeipmi/ipckey\" -DIPMI_DEBUG_IPCKEY=\"../libfreeipmi/driver/ipmi-semaphores.h\" -D_REENTRANT -g -O2 -c sel/ipmi-sel-string-quanta-s99q.c  -fPIC -DPIC -o sel/.libs/libfreeipmi_la-ipmi-sel-string-quanta-s99q.o
       > /nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../config  -I../common/debugutil -I../common/miscutil -I../common/portability -I../libfreeipmi/include -I../libfreeipmi/include -I../libfreeipmi -DIPMI_IPCKEY=\"/nix/store/0lnd5a6imppj11arqipq6vpkkk90ish4-freeipmi-1.6.14/var/lib/freeipmi/ipckey\" -DIPMI_DEBUG_IPCKEY=\"../libfreeipmi/driver/ipmi-semaphores.h\" -D_REENTRANT   -g -O2 -c -o sel/libfreeipmi_la-ipmi-sel-string-sun-x4140.lo `test -f 'sel/ipmi-sel-string-sun-x4140.c' || echo './'`sel/ipmi-sel-string-sun-x4140.c
       > libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../config -I../common/debugutil -I../common/miscutil -I../common/portability -I../libfreeipmi/include -I../libfreeipmi/include -I../libfreeipmi -DIPMI_IPCKEY=\"/nix/store/0lnd5a6imppj11arqipq6vpkkk90ish4-freeipmi-1.6.14/var/lib/freeipmi/ipckey\" -DIPMI_DEBUG_IPCKEY=\"../libfreeipmi/driver/ipmi-semaphores.h\" -D_REENTRANT -g -O2 -c sel/ipmi-sel-string-sun-x4140.c  -fPIC -DPIC -o sel/.libs/libfreeipmi_la-ipmi-sel-string-sun-x4140.o
       > /nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../config  -I../common/debugutil -I../common/miscutil -I../common/portability -I../libfreeipmi/include -I../libfreeipmi/include -I../libfreeipmi -DIPMI_IPCKEY=\"/nix/store/0lnd5a6imppj11arqipq6vpkkk90ish4-freeipmi-1.6.14/var/lib/freeipmi/ipckey\" -DIPMI_DEBUG_IPCKEY=\"../libfreeipmi/driver/ipmi-semaphores.h\" -D_REENTRANT   -g -O2 -c -o sel/libfreeipmi_la-ipmi-sel-string-supermicro-common.lo `test -f 'sel/ipmi-sel-string-supermicro-common.c' || echo './'`sel/ipmi-sel-string-supermicro-common.c
       > libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../config -I../common/debugutil -I../common/miscutil -I../common/portability -I../libfreeipmi/include -I../libfreeipmi/include -I../libfreeipmi -DIPMI_IPCKEY=\"/nix/store/0lnd5a6imppj11arqipq6vpkkk90ish4-freeipmi-1.6.14/var/lib/freeipmi/ipckey\" -DIPMI_DEBUG_IPCKEY=\"../libfreeipmi/driver/ipmi-semaphores.h\" -D_REENTRANT -g -O2 -c sel/ipmi-sel-string-supermicro-common.c  -fPIC -DPIC -o sel/.libs/libfreeipmi_la-ipmi-sel-string-supermicro-common.o
       > /nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../config  -I../common/debugutil -I../common/miscutil -I../common/portability -I../libfreeipmi/include -I../libfreeipmi/include -I../libfreeipmi -DIPMI_IPCKEY=\"/nix/store/0lnd5a6imppj11arqipq6vpkkk90ish4-freeipmi-1.6.14/var/lib/freeipmi/ipckey\" -DIPMI_DEBUG_IPCKEY=\"../libfreeipmi/driver/ipmi-semaphores.h\" -D_REENTRANT   -g -O2 -c -o sel/libfreeipmi_la-ipmi-sel-string-supermicro-overheat-common.lo `test -f 'sel/ipmi-sel-string-supermicro-overheat-common.c' || echo './'`sel/ipmi-sel-string-supermicro-overheat-common.c
       > libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../config -I../common/debugutil -I../common/miscutil -I../common/portability -I../libfreeipmi/include -I../libfreeipmi/include -I../libfreeipmi -DIPMI_IPCKEY=\"/nix/store/0lnd5a6imppj11arqipq6vpkkk90ish4-freeipmi-1.6.14/var/lib/freeipmi/ipckey\" -DIPMI_DEBUG_IPCKEY=\"../libfreeipmi/driver/ipmi-semaphores.h\" -D_REENTRANT -g -O2 -c sel/ipmi-sel-string-supermicro-overheat-common.c  -fPIC -DPIC -o sel/.libs/libfreeipmi_la-ipmi-sel-string-supermicro-overheat-common.o
       > /nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../config  -I../common/debugutil -I../common/miscutil -I../common/portability -I../libfreeipmi/include -I../libfreeipmi/include -I../libfreeipmi -DIPMI_IPCKEY=\"/nix/store/0lnd5a6imppj11arqipq6vpkkk90ish4-freeipmi-1.6.14/var/lib/freeipmi/ipckey\" -DIPMI_DEBUG_IPCKEY=\"../libfreeipmi/driver/ipmi-semaphores.h\" -D_REENTRANT   -g -O2 -c -o sel/libfreeipmi_la-ipmi-sel-string-supermicro-magnum-technologies-x8dtl.lo `test -f 'sel/ipmi-sel-string-supermicro-magnum-technologies-x8dtl.c' || echo './'`sel/ipmi-sel-string-supermicro-magnum-technologies-x8dtl.c
       > libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../config -I../common/debugutil -I../common/miscutil -I../common/portability -I../libfreeipmi/include -I../libfreeipmi/include -I../libfreeipmi -DIPMI_IPCKEY=\"/nix/store/0lnd5a6imppj11arqipq6vpkkk90ish4-freeipmi-1.6.14/var/lib/freeipmi/ipckey\" -DIPMI_DEBUG_IPCKEY=\"../libfreeipmi/driver/ipmi-semaphores.h\" -D_REENTRANT -g -O2 -c sel/ipmi-sel-string-supermicro-magnum-technologies-x8dtl.c  -fPIC -DPIC -o sel/.libs/libfreeipmi_la-ipmi-sel-string-supermicro-magnum-technologies-x8dtl.o
       > /nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../config  -I../common/debugutil -I../common/miscutil -I../common/portability -I../libfreeipmi/include -I../libfreeipmi/include -I../libfreeipmi -DIPMI_IPCKEY=\"/nix/store/0lnd5a6imppj11arqipq6vpkkk90ish4-freeipmi-1.6.14/var/lib/freeipmi/ipckey\" -DIPMI_DEBUG_IPCKEY=\"../libfreeipmi/driver/ipmi-semaphores.h\" -D_REENTRANT   -g -O2 -c -o sel/libfreeipmi_la-ipmi-sel-string-supermicro-peppercon-common.lo `test -f 'sel/ipmi-sel-string-supermicro-peppercon-common.c' || echo './'`sel/ipmi-sel-string-supermicro-peppercon-common.c
       > libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../config -I../common/debugutil -I../common/miscutil -I../common/portability -I../libfreeipmi/include -I../libfreeipmi/include -I../libfreeipmi -DIPMI_IPCKEY=\"/nix/store/0lnd5a6imppj11arqipq6vpkkk90ish4-freeipmi-1.6.14/var/lib/freeipmi/ipckey\" -DIPMI_DEBUG_IPCKEY=\"../libfreeipmi/driver/ipmi-semaphores.h\" -D_REENTRANT -g -O2 -c sel/ipmi-sel-string-supermicro-peppercon-common.c  -fPIC -DPIC -o sel/.libs/libfreeipmi_la-ipmi-sel-string-supermicro-peppercon-common.o
       > /nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../config  -I../common/debugutil -I../common/miscutil -I../common/portability -I../libfreeipmi/include -I../libfreeipmi/include -I../libfreeipmi -DIPMI_IPCKEY=\"/nix/store/0lnd5a6imppj11arqipq6vpkkk90ish4-freeipmi-1.6.14/var/lib/freeipmi/ipckey\" -DIPMI_DEBUG_IPCKEY=\"../libfreeipmi/driver/ipmi-semaphores.h\" -D_REENTRANT   -g -O2 -c -o sel/libfreeipmi_la-ipmi-sel-string-supermicro-x10dimm-common.lo `test -f 'sel/ipmi-sel-string-supermicro-x10dimm-common.c' || echo './'`sel/ipmi-sel-string-supermicro-x10dimm-common.c
       > libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../config -I../common/debugutil -I../common/miscutil -I../common/portability -I../libfreeipmi/include -I../libfreeipmi/include -I../libfreeipmi -DIPMI_IPCKEY=\"/nix/store/0lnd5a6imppj11arqipq6vpkkk90ish4-freeipmi-1.6.14/var/lib/freeipmi/ipckey\" -DIPMI_DEBUG_IPCKEY=\"../libfreeipmi/driver/ipmi-semaphores.h\" -D_REENTRANT -g -O2 -c sel/ipmi-sel-string-supermicro-x10dimm-common.c  -fPIC -DPIC -o sel/.libs/libfreeipmi_la-ipmi-sel-string-supermicro-x10dimm-common.o
       > sel/ipmi-sel-string-supermicro-x10dimm-common.c: In function 'sel_string_output_supermicro_x10dimm_event_data2_event_data3':
       > sel/ipmi-sel-string-supermicro-x10dimm-common.c:87:14: error: implicit declaration of function 'sel_string_output_supermicro_dimm_event_data2_event_data3'; did you mean 'sel_string_output_supermicro_x10dimm_event_data2_event_data3'? []
       >    87 |   if ((ret = sel_string_output_supermicro_dimm_event_data2_event_data3 (ctx,
       >       |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       >       |              sel_string_output_supermicro_x10dimm_event_data2_event_data3
       > make[2]: *** [Makefile:2700: sel/libfreeipmi_la-ipmi-sel-string-supermicro-x10dimm-common.lo] Error 1
       > make[2]: Leaving directory '/build/freeipmi-1.6.14/libfreeipmi'
       > make[1]: *** [Makefile:3005: all-recursive] Error 1
       > make[1]: Leaving directory '/build/freeipmi-1.6.14/libfreeipmi'
       > make: *** [Makefile:602: all-recursive] Error 1
       For full logs, run 'nix log /nix/store/8f4qj079jqpsp420hmf0slqlhdb58xnb-freeipmi-1.6.14.drv'.
error: 1 dependencies of derivation '/nix/store/9adizdyjl3n66cj99xvb3gnfafafx5ji-hardcode-paths.patch.drv' failed to build
```


## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

## Notify maintainers

@raskin
@pierron

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/fr/freeipmi/package.nix
+++ b/pkgs/by-name/fr/freeipmi/package.nix
@@ -30,6 +30,10 @@ stdenv.mkDerivation rec {
     "ac_cv_file__dev_random=true"
   ];
 
+  # Fix GCC 14 build.
+  # https://savannah.gnu.org/bugs/?65203
+  env.NIX_CFLAGS_COMPILE = "-Wno-error=implicit-function-declaration";
+
   doCheck = true;
 
   meta = {

